drivers/spi/spi_nrfx_spim.c

```diff
@@ -74,10 +88,55 @@ struct spi_nrfx_config {
 #ifdef CONFIG_DCACHE
 	uint32_t mem_attr;
 #endif
+#ifdef USE_CLOCK_REQUESTS
+	const struct device *clk_dev;
+	struct nrf_clock_spec clk_spec;
+#endif
 };
 
 static void event_handler(const nrfx_spim_evt_t *p_event, void *p_context);
 
+static inline int request_clock(const struct device *dev)
```

Comment:
won't the compiler complain about unused `dev` parameter when `USE_CLOCK_REQUESTS` is undefined? same for `release_clock`

Reply:
Yes, I think it may complain, depending on the enabled warnings. I'll add `ARG_UNUSED()` macro calls.

Reply:
Added.